cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiff.java

```diff
@@ -0,0 +1,214 @@
+                        log.trace("{} not set in desired, using default value", entry.get().name());
+                    } else {
+                        // entry is in current, is not in desired, is not default -> it was using non-default value and was removed
+                        // if the entry was custom, it should be deleted
+                        if (!isIgnorableProperty(pathValueWithoutSlash)) {
+                            difference.put(pathValueWithoutSlash, null);
+                            updatedCE.add(new AlterConfigOp(new ConfigEntry(pathValueWithoutSlash, null), AlterConfigOp.OpType.DELETE));
+                            log.trace("{} not set in desired, unsetting back to default {}", entry.get().name(), null);
+                        }
+                    }
+                } else if (d.get("op").asText().equals("replace")) {
+                    // entry is in the current, desired is updated value
+                    if (!isIgnorableProperty(pathValueWithoutSlash)) {
+                        log.trace("{} has new desired value {}", entry.get().name(), desiredMap.get(entry.get().name()));
+                        updatedCE.add(new AlterConfigOp(new ConfigEntry(pathValueWithoutSlash, desiredMap.get(pathValueWithoutSlash)), AlterConfigOp.OpType.SET));
+                        difference.put(pathValueWithoutSlash, desiredMap.get(pathValueWithoutSlash));
+                    }
+                }
+            } else {
+                if (d.get("op").asText().equals("add")) {
+                    // entry is not in the current, it is added
+                    if (!isIgnorableProperty(pathValueWithoutSlash)) {
+                        log.trace("add new {} {}", pathValueWithoutSlash, d.get("op").asText());
+                        difference.put(pathValueWithoutSlash, desiredMap.get(pathValueWithoutSlash));
+                        updatedCE.add(new AlterConfigOp(new ConfigEntry(pathValueWithoutSlash, desiredMap.get(pathValueWithoutSlash)), AlterConfigOp.OpType.SET));
+                    }
+                }
+            }
+
+            log.debug("Kafka Broker Config Differs : {}", d);
+            log.debug("Current Kafka Broker Config path {} has value {}", pathValueWithoutSlash, lookupPath(source, pathValue));
+            log.debug("Desired Kafka Broker Config path {} has value {}", pathValueWithoutSlash, lookupPath(target, pathValue));
+        }
+
+
+        difference.entrySet().forEach(e -> {
+            log.info("{} broker conf differs: '{}' -> '{}'", e.getKey(), currentMap.get(e.getKey()), e.getValue());
+        });
+
+        updated.put(new ConfigResource(ConfigResource.Type.BROKER, Integer.toString(brokerId)), updatedCE);
+
+        String diffString = difference.toString();
+        diffString = diffString.substring(1, diffString.length() - 1).replace(", ", "\n");
+        return KafkaConfiguration.unvalidated(diffString);
```

Comment:
This is ugly. Why serialize to a String only to parse it back into a map (which is what KafkaConfiguration is doing)? If you need a `KafkaConfiguration.unvalidated(Map<String, String>)` then write one!